TESTS/mbed_drivers/lp_ticker/main.cpp

```diff
@@ -91,9 +95,12 @@ void test_multi_ticker(void)
     }
 
     Thread::wait(MULTI_TICKER_TIME_MS + TICKER_COUNT + extra_wait);
+    multi_counter_snapshot = multi_counter;
```

Comment:
> When very early interrupt is scheduled (e.g now + 2[ticks]) then it's likely that it won't be fired in some circumstances and as a result there is overdue event in queue. That overdue event can be mistakenly scheduled again by Ticker::detach (detach calls schedule if head was removed). That's why we should check interrupts counter immediately after wait period and before detach loop

I don't follow the description and these changes here. can you elaborate how multicounter snapshot fixes it?

Reply:
Assume the following scenario (time is mesured in ticks)
Running test_multi_ticker with `TICKER_COUNT == 3` (tickers t1, t2, t3) and ticker period 100
Then we get following events scheduled at time 0 

 event_queue: e1(100), e2(105), e3(105)
 now == 0
 
 at time 100 e1 is triggered (and next event for t1 is scheduled e11(200))
 at time 103 e2 interrupt is set(scheduled) (now == 103)
 for e2 interrupt will not be trigerred (trying to set (now + 2) what is less then (now + 3))
 (on some targets to be sure that interrupt will be triggered we can't set interrupt earlier than (now + 3 ticks))
 
 
 After wait instruction we have following state:
 
 event_queue: e2(105), e3(105), e11(200)
 `multi_counter == 1` (e1 triggered)
 now == 110
 `multi_counter_snapshot = multi_counter`

 In detach loop:
 detach t1 events then e11 will be removed
 detach t2 events then e2 will be removed 
			at this point since head was removed, 
                        `schedule_interrupt` is called and since e3 
			is overdue `fire_interrupt` is called for e3
                        (and next event for t3 is scheduled e31(205))
 detach t3 events then e31(205) will be removed 
 
 
 After detach loop we have following state:
 
 event_queue: empty
 `multi_counter == 2` (e1 and e3 triggered)
 now == 115
 
 result when checking `multi_counter`: expected 3 is 2
 result when checking `multi_counter_snapshot`: expected 3 is 1
 
 So using `multi_counter_snapshot` we catch real test status, just after test end